cpp/src/arrow/util/bit_block_counter.h

```diff
@@ -292,9 +413,10 @@ class ARROW_EXPORT OptionalBinaryBitBlockCounter {
 // Functional-style bit block visitors.
 
 template <typename VisitNotNull, typename VisitNull>
-Status VisitBitBlocks(const std::shared_ptr<Buffer>& bitmap_buf, int64_t offset,
-                      int64_t length, VisitNotNull&& visit_not_null,
-                      VisitNull&& visit_null) {
+static inline Status VisitBitBlocks(const std::shared_ptr<Buffer>& bitmap_buf,
```

Comment:
Ah, my bad. What about `static`, do you think it is useful?
(my thought: it doesn't expose the generated specializations as global symbols)

Reply:
SGTM.

I was actually surprised to learn that this has an effect:
- When linking object files which each have `inline void noop(){}` the symbol in the resulting shared object will be non-global so it can be `strip`'d out. The same is true for
  ```
  static void noop(){}
  __attribute__((visibility("hidden"))) void noop(){}
  namespace{  void noop(){}  }
  ```
- When linking object files which each have `template <typename T> void noop(){}` the symbol will be weak but *global*, so it survives `strip`. By contrast the symbol for `template <typename T> static void noop(){}` will be non-global.